src/java/org/apache/cassandra/service/ConsensusTableMigrationState.java

```diff
@@ -0,0 +1,697 @@
+        {
+            // Only successes drive forward progress
+        }
+    };
+
+    @VisibleForTesting
+    public static void reset()
+    {
+        ClusterMetadataService.reset();
+    }
+
+    public enum ConsensusMigrationRepairType
+    {
+        ineligible(0),
+        paxos(1),
+        accord(2);
+
+        public final byte value;
+
+        ConsensusMigrationRepairType(int value)
+        {
+            this.value = SignedBytes.checkedCast(value);
+        }
+
+        public static ConsensusMigrationRepairType fromString(String repairType)
+        {
+            return ConsensusMigrationRepairType.valueOf(repairType.toLowerCase());
+        }
+
+        public static ConsensusMigrationRepairType fromValue(byte value)
+        {
+            switch (value)
+            {
+                default:
+                    throw new IllegalArgumentException(value + " is not recognized");
+                case 0:
+                    return ConsensusMigrationRepairType.paxos;
+                case 1:
+                    return ConsensusMigrationRepairType.accord;
+            }
+        }
+    }
+
+    public enum ConsensusMigrationTarget {
+        paxos(0),
+        accord(1);
+
+        public final byte value;
+
+        ConsensusMigrationTarget(int value)
+        {
+            this.value = SignedBytes.checkedCast(value);
+        }
+
+        public static ConsensusMigrationTarget fromString(String targetProtocol)
+        {
+            return ConsensusMigrationTarget.valueOf(targetProtocol.toLowerCase());
+        }
+
+        public static ConsensusMigrationTarget fromValue(byte value)
+        {
+            switch (value)
+            {
+                default:
+                    throw new IllegalArgumentException(value + " is not recognized");
+                case 0:
+                    return paxos;
+                case 1:
+                    return accord;
+            }
+        }
+    }
+
+    public static class ConsensusMigrationRepairResult
+    {
+        private final ConsensusMigrationRepairType consensusMigrationRepairType;
+        private final Epoch minEpoch;
+
+        private ConsensusMigrationRepairResult(ConsensusMigrationRepairType consensusMigrationRepairType, Epoch minEpoch)
+        {
+            this.consensusMigrationRepairType = consensusMigrationRepairType;
+            this.minEpoch = minEpoch;
+        }
+
+        public static ConsensusMigrationRepairResult fromCassandraRepair(Epoch minEpoch, boolean didPaxosAndRegularRepair)
+        {
+            checkArgument(!didPaxosAndRegularRepair || minEpoch.isAfter(Epoch.EMPTY), "Epoch should not be empty if Paxos and regular repairs were performed");
+            if (didPaxosAndRegularRepair)
+                return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.paxos, minEpoch);
+            else
+                return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.ineligible, Epoch.EMPTY);
```

Comment:
a repair should also be ineligible if it was run with `--force` and repair then skipped one or more nodes

Reply:
Added this behavior and a test